src/main/scala/org/opennetworkinsight/ProxyPostLDA.scala

```diff
@@ -1,60 +1,37 @@
 package org.opennetworkinsight
 
 import org.apache.log4j.{Level, Logger => ApacheLogger}
-import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
 import org.apache.spark.storage.StorageLevel
 import org.apache.spark.SparkContext
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.functions._
 import org.slf4j.{Logger, LoggerFactory}
 
 /**
   *
   */
 object ProxyPostLDA {
 
-  def getColumnNames(input: Array[String], sep: Char = ','): scala.collection.mutable.Map[String, Int] = {
-    val columns = scala.collection.mutable.Map[String, Int]()
-    val header = input.zipWithIndex
-    header.foreach(tuple => columns(tuple._1) = tuple._2)
-    columns
-  }
-
-  def proxyPostLDA(inputPath: String, resultsFilePath: String, threshold: Double, documentResults: Array[String],
-                 wordResults: Array[String], sc: SparkContext, sqlContext: SQLContext, logger: Logger) = {
+  def getResults(inputPath: String, resultsFilePath: String, topicCount: Int, threshold: Double,
+                 documentResults: Array[String],  wordResults: Array[String],
+                 sc: SparkContext, sqlContext: SQLContext, logger: Logger) = {
 
     logger.info("Proxy post LDA starts")
 
-    val topics_lines = documentResults
-    val words_lines = wordResults
-
-    val l_topics = topics_lines.map(line => {
-      val ip = line.split(",")(0)
-      val topicProbs = line.split(",")(1).split(' ').map(_.toDouble)
-      (ip, topicProbs)
-    }).map({case (ip, topicProbs) => ip -> topicProbs }).toMap
-
-    val topics = sc.broadcast(l_topics)
+    val ipToTopicMix : Map[String, Array[Double]] = getIpToTopicMix(documentResults)
 
-    val l_words  = words_lines.map(line => {
+    val wordsToProbPerTopic : Map[String, Array[Double]]   = wordResults.map(line => {
       val word = line.split(",")(0)
       val probPerTopic = line.split(",")(1).split(' ').map(_.toDouble)
       (word, probPerTopic)
     }).map({case (word, probPerTopic)  => word -> probPerTopic}).toMap
 
-    val words = sc.broadcast(l_words)
-
-    logger.info("Proxy post LDA starts")
 
-    var df_cols = new Array[String](0)
-
-    val multidata = {
-      var df = sqlContext.parquetFile(inputPath.split(",")(0)).filter("proxy_date is not null and proxy_time is not null and proxy_clientip is not null")
-      val files = inputPath.split(",")
-      for ((file, index) <- files.zipWithIndex) {
-        if (index > 1) {
-          df = df.unionAll(sqlContext.parquetFile(file).filter("proxy_date is not null and proxy_time is not null and proxy_clientip is not null"))
-        }
-      }
-      df = df.select("proxy_date",
+    val dataFrame = sqlContext.parquetFile(inputPath)
```

Comment:
I think dataFrame is very generic, I like the ones like scoredDF or filteredDf. Wouldn't be better something like rawDataDF?